numpy/random/mtrand/mtrand.pyx

```diff
@@ -681,9 +681,14 @@ cdef class RandomState:
                 with self.lock:
                     rk_seed(idx, self.internal_state)
         except TypeError:
-            obj = np.asarray(seed).astype(np.int64, casting='safe')
-            if ((obj > int(2**32 - 1)) | (obj < 0)).any():
-                raise ValueError("Seed must be between 0 and 2**32 - 1")
+            obj = np.asarray(seed)
+            if obj.size == 0:
+                raise ValueError("Seed must convertible to a 32-bit unsigned "
+                                 "integer and non-empty")
```

Comment:
"convertible to a 32-bit unsigned integer" doesn't seem relevant to the problem here

Reply:
It's just a way of restating the original message and emphasizing that the values must be integral rather than just floats in that range.